contracts/SpokePoolV3Periphery.sol

```diff
@@ -289,323 +377,471 @@ contract SpokePoolV3Periphery is Lockable, MultiCaller {
-        uint256 inputTokenBalanceBefore,
-        uint256 minExpectedInputTokenAmount,
-        DepositData calldata depositData,
-        IERC20 _swapToken,
-        IERC20 _acrossInputToken
-    ) private {
         // Sanity check that we received as many tokens as we require:
-        uint256 returnAmount = _acrossInputToken.balanceOf(address(this)) - inputTokenBalanceBefore;
+        uint256 returnAmount = _acrossInputToken.balanceOf(address(this)) - dstBalanceBefore;
+
         // Sanity check that received amount from swap is enough to submit Across deposit with.
-        if (returnAmount < minExpectedInputTokenAmount) revert MinimumExpectedInputAmount();
+        if (returnAmount < swapAndDepositData.minExpectedInputTokenAmount) revert MinimumExpectedInputAmount();
         // Sanity check that we don't have any leftover swap tokens that would be locked in this contract (i.e. check
         // that we weren't partial filled).
-        if (swapTokenBalanceBefore - _swapToken.balanceOf(address(this)) != swapTokenAmount) revert LeftoverSrcTokens();
+        if (srcBalanceBefore - _swapToken.balanceOf(address(this)) != _swapTokenAmount) revert LeftoverSrcTokens();
 
         emit SwapBeforeBridge(
-            exchange,
-            routerCalldata,
+            _exchange,
+            swapAndDepositData.routerCalldata,
             address(_swapToken),
             address(_acrossInputToken),
-            swapTokenAmount,
+            _swapTokenAmount,
             returnAmount,
-            depositData.outputToken,
-            depositData.outputAmount
+            swapAndDepositData.depositData.outputToken,
+            swapAndDepositData.depositData.outputAmount
         );
+
         // Deposit the swapped tokens into Across and bridge them using remainder of input params.
-        _depositV3(_acrossInputToken, returnAmount, depositData);
+        _depositV3(
+            swapAndDepositData.depositData.depositor,
+            swapAndDepositData.depositData.recipient,
+            address(_acrossInputToken),
+            swapAndDepositData.depositData.outputToken,
+            returnAmount,
+            swapAndDepositData.depositData.outputAmount,
+            swapAndDepositData.depositData.destinationChainId,
+            swapAndDepositData.depositData.exclusiveRelayer,
+            swapAndDepositData.depositData.quoteTimestamp,
+            swapAndDepositData.depositData.fillDeadline,
+            swapAndDepositData.depositData.exclusivityParameter,
+            swapAndDepositData.depositData.message
+        );
     }
 
+    /**
+     * @notice Creates the EIP712 compliant hashed data corresponding to the BaseDepositData struct.
+     * @param baseDepositData Input struct whose values are hashed.
+     * @dev BaseDepositData is only used as a nested struct for both DepositData and SwapAndDepositData.
+     */
+    function _hashBaseDepositData(BaseDepositData calldata baseDepositData) private pure returns (bytes32) {
```

Comment:
Thoughts on moving these hashing functions and `DepositData/BaseDepositData/SwapAndDepositData` and all EIP712 stuff to a library? Something like PeripherySigningLib?

Reply:
yes please!

Reply:
This contract is getting too big

Reply:
Library added here: https://github.com/across-protocol/contracts/pull/782/commits/e1162402065398e41221f13ffb2a25a39f2d03ef

Reply:
I wouldn't have moved the Swap/Deposit data into the "Signing" lib, i would have kept those in the Periphery where they are used. Just a nit

Reply:
I'll change it.